apps/files_sharing/lib/Controller/ShareesAPIController.php

```diff
@@ -197,6 +197,27 @@ public function search(string $search = '', ?string $itemType = null, int $page
 			$result['exact'] = array_merge($this->result['exact'], $result['exact']);
 		}
 		$this->result = array_merge($this->result, $result);
+
+		// Get a list of groups to exclude from search results
+		$blacklisted_groups = $this->config->getSystemValue('blacklisted_groups', true);
```

Comment:
```suggestion
		// Get a list of groups to exclude from search results
		$blacklisted_groups = $this->config->getSystemValue('blacklisted_groups', []);
```
The default value needs to be an empty array, as otherwise the later code will fail if the config is not set.